src/psyclone/psyGen.py

```diff
@@ -4365,3 +4369,13 @@ def update(self):
         self._pre_gen_validate()
         self._add_region(start_text="DATA", end_text="END DATA",
                          data_movement="analyse")
+
+
+__all__ = ['PSyFactory', 'PSy', 'Invokes', 'Invoke', 'InvokeSchedule',
```

Comment:
Thanks very much for adding all of these - above and beyond! Please could you add a comment explaining that this is for automatic documentation generation using Sphinx AutoAPI.

Reply:
Actually, I've just realised that there are now 5 new warnings when building the Reference Guide:

    /home/Projects/PSyclone/src/psyclone/psyGen.py:docstring of psyclone.psyGen.HaloExchange:7: WARNING: Field list ends without a blank line; unexpected unindent.
    /home/Projects/PSyclone/src/psyclone/psyGen.py:docstring of psyclone.psyGen.HaloExchange.check_vector_halos_differ:10: WARNING: Field list ends without a blank line; unexpected unindent.
    /home/Projects/PSyclone/src/psyclone/psyGen.py:docstring of psyclone.psyGen.CodedKern.gen_code:5: WARNING: Unknown interpreted text role "py:calls".
    /home/Projects/PSyclone/src/psyclone/psyGen.py:docstring of psyclone.psyGen.Argument.backward_write_dependencies:7: WARNING: Inline interpreted text or phrase reference start-string without end-string.
    /home/Projects/PSyclone/src/psyclone/psyGen.py:docstring of psyclone.psyGen.Invoke:5: WARNING: Unknown target name: "invoke".

Please could fix these.

Reply:
Comment added and warnings should be gone now.